stratify/_vinterp.pyx

```diff
@@ -272,6 +272,28 @@ cdef long nearest_edge_extrap(int direction, double[:] z_src,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+cdef long linear_extrap(int direction, double[:] z_src,
+                        double[:, :] fz_src, double level,
+                        double[:] fz_target) nogil except -1:
+    """Linear extrapolation using either the first or last 2 values."""
+    cdef unsigned int m = fz_src.shape[0]
+    cdef unsigned int p0, p1, i
+    cdef double frac
+
+    if direction < 0:
```

Comment:
When fixing this I had two options - raise an exception, or to simply return the single fz value for any fz_target (i.e. linear becomes nearest). I went for the exception.